### ISSUE TITLE
quick question about an implementation detail

### ISSUE THREAD

**Issue by lwitzani** [oiva/homebridge-air-quality-file#5]
Hi, first of all thanks so much for the great working plugin!

I just wondered why you would use all available data readings in the following spot:
```
 getAirQuality(readings: Reading[]): number {
    const totalPM25 = readings.reduce(
      (prev, reading): number => prev + parseFloat(reading.pm25),
      0
    )
    const totalPM10 = readings.reduce(
      (prev, reading): number => prev + parseFloat(reading.pm10),
      0
    )

    const averagePM25 = totalPM25 / readings.length
    const averagePM10 = totalPM10 / readings.length
```

I increased my number of readings in my json file to like 300 or so und my readings are every 10 minutes. So this code calculates the average over all 300 (very old) readings although actually only the latest one would be interesting for the current air quality. 

Do you remember why you implemented it that way? 

Thanks :)

**Comment by oiva** [comment]
Hi and sorry for not getting back to you sooner. The code is a bit hard to follow (I had to re-read it a couple times to see what's going on 😬) but the readings that end up in `getAirQuality` are already filtered to only include readings inside the timeframe defined by the 'durationToAverage' setting.

The execution goes like `updateAirQualityIndex` -> `readFile` -> `parseData` -> `filterReadings` (which throws everything out that's older than 'current time minus durationToAverage'). The resulting data is then passed to `getAirQuality` (in `readFile` callback) where all of the data is used to calculate the average air quality during the 'durationToAverage'.

Looking at it now the code would probably be more readable if the `readFile` function only... _read_ the file and parsing and filtering the data happened in the callback. Is this something you would like to refactor a bit?

And there might of course be a bug somewhere in the chain in which case I need to read it once more!

**Comment by lwitzani** [comment]
Oh :) okay thx so i did actually misinterpreted the code then, not seeing that the readings are actually filtered before. So then it makes perfectly sense (configuring a timeframe and calculating an average in that time frame). At the time of creating this issue, i did not 100% understand the code since i only looked at it using a text editor 😅 sry

cheers
